packages/did-comm/src/types/IDIDComm.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  IAgentContext,
+  IDIDManager,
+  IKeyManager,
+  IMessage,
+  IMessageHandler,
+  IPluginMethodMap,
+  IResolver,
+} from '@veramo/core'
+import {
+  IPackDIDCommMessageArgs,
+  ISendDIDCommMessageArgs,
+  ISendDIDCommMessageResult,
+  ISendMessageDIDCommAlpha1Args,
+  IUnpackDIDCommMessageArgs,
+} from '../action-handler'
+import { DIDCommMessageMediaType, IPackedDIDCommMessage, IUnpackedDIDCommMessage } from './message-types'
+
+/**
+ * DID Comm plugin interface for {@link @veramo/core#Agent}
+ * @beta
+ */
+export interface IDIDComm extends IPluginMethodMap {
+  /**
+   * Partially decodes a possible DIDComm message string to determine the {@link DIDCommMessageMediaType}
+   *
+   * @param IPackedDIDCommMessage - the message to be interpreted
+   * @returns the {@link DIDCommMessageMediaType} if it was successfully parsed
+   * @throws if the message cannot be parsed as DIDComm v2
+   *
+   * @beta
+   */
+  getDIDCommMessageMediaType(args: IUnpackDIDCommMessageArgs): Promise<DIDCommMessageMediaType>
```

Comment:
`IUnpackDIDCommMessageArgs` seems to not match the param type.

Reply:
`IUnpackDIDCommMessageArgs` is `{ message: string }`
I did not want to define a new virtually identical `*Args` type, especially since this method might be removed from the public plugin API.

Reply:
right, but would `IPackedDIDCommMessage` make more sense?

Reply:
actually, yes. That's a good suggestion.